Source/Applications/ApplicationArtifactIdentifierStringConverter.cs

```diff
@@ -129,6 +136,9 @@ void ValidateIdentifierString(string identifierAsString)
             var applicationArtifactTypeSeparatorIndex = identifierAsString.IndexOf(ApplicationArtifactTypeSeparator);
             ThrowIfApplicationArtifactTypeMissing(applicationArtifactTypeSeparatorIndex, identifierAsString);
 
+            var applicationArtifactGenerationSeperatorIndex = identifierAsString.LastIndexOf(ApplicationArtifactGenerationSeperator);
```

Comment:
If the character is unique, these checks becomes simpler. If we were to add another using the same separator - this validation increases in complexity